src/mds/CDir.cc

```diff
@@ -218,29 +218,43 @@ CDir::CDir(CInode *in, frag_t fg, MDCache *mdcache, bool auth) :
  * If mds_debug_scatterstat is enabled, assert for correctness,
  * otherwise just print out the mismatch and continue.
  */
-bool CDir::check_rstats()
+bool CDir::check_rstats(bool scrub)
```

Comment:
This scrub parameter confused me a bit, both in name and purpose. I think it might be clearer if we split it into two different flags. "force" (which forces a check even if mds_debug_scatterstat is disabled), and "assert_failures" which covers whether we assert that everything matches up.